### ISSUE TITLE
1.40.0: pytest fails with deprecation errors

### ISSUE THREAD

**Issue by kloczek** [getsentry/sentry-python#2693]
### How do you use Sentry?

Sentry Saas (sentry.io)

### Version

1.40.0

### Steps to Reproduce

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix> using 'installer` module
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

### Expected Result

pytest should not fail

### Actual Result

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-sentry-1.40.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-sentry-1.40.0-2.fc35.x86_64/usr/
lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/sentry-python-1.40.0/tests/conftest.py'.
tests/__init__.py:12: in <module>
    _warning_recorder_mgr = pytest.warns(None)
E   pytest.PytestRemovedIn8Warning: Passing None has been deprecated.
E   See https://docs.pytest.org/en/latest/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests for alternatives in common use cases.
```

**Comment by sentrivana** [comment]
Hey @kloczek, thanks for reporting this.

This happens if you run the tests on with a newer `pytest` version. We're using a [pinned version of `tox`](https://github.com/getsentry/sentry-python/blob/c8e917205d77e143c68c0cda4df877167d402830/test-requirements.txt#L8) to run our test suite, which, in turn, [pins `pytest`](https://github.com/tox-dev/tox/blob/f9664f200ba1298fb78a4d1405b7910aa62fe0f7/setup.py#L44) to `>= 3.0.0, <4`. Is it an option for you to run the test suite a `pytest` version from that range?

We have [plans for updating our tooling](https://github.com/getsentry/sentry-python/issues/2586) in the next month or so, which might also include removing the pin on the (pretty ancient) `tox` version, but this will still take a bit, so in the meantime, the safest bet is to use the same `pytest` version we're using.

**Comment by kloczek** [comment]
> This happens if you run the tests on with a newer `pytest` version. We're using a [pinned version of `tox`](https://github.com/getsentry/sentry-python/blob/c8e917205d77e143c68c0cda4df877167d402830/test-requirements.txt#L8) to run our test suite, which, in turn, [pins `pytest`](https://github.com/tox-dev/tox/blob/f9664f200ba1298fb78a4d1405b7910aa62fe0f7/setup.py#L44) to `>= 3.0.0, <4`. Is it an option for you to run the test suite a `pytest` version from that range?

Sorry I'm not going to waste time on one package with python module because it still requires some ancient version of pytest (last pytest 3.x has been released +5 years ago and latest version 8.0.0 has been released week ago).

**Comment by szokeasaurusrex** [comment]
> Sorry I'm not going to waste time on one package with python module because it still requires some ancient version of pytest (last pytest 3.x has been released +5 years ago and latest version 8.0.0 has been released week ago).

Hi @kloczek, I completely understand your frustration here. I am investigating this problem, and I will let you know as soon as I have a fix, or at least, a better workaround.

**Comment by sentrivana** [comment]
Just a correction, my bad -- the `tox` version we're using doesn't actually have a restriction on `pytest`; only the `tox[testing]` extra (which we don't use). We're still using an older `pytest` version though.

**Comment by szokeasaurusrex** [comment]
@kloczek This problem is fixed by #2712.

Until #2712 is released, you can work around the issue by using `pytest` <8.0.0. `pytest` 7.x.x should work fine for this purpose; the `>= 3.0.0, <4` requirement @sentrivana referred to applies only to the tox "testing" extra that we don't use. We use `pytest==6.2.5` in our Python 3.8 tests, so if for any reason you struggle with 7.x.x, you can try 6.x.x
